src/main/java/com/zaxxer/nuprocess/osx/ProcessKqueue.java

```diff
@@ -84,15 +94,52 @@ public void registerProcess(OsxProcess process)
 
       pidToProcessMap.put(pid, process);
 
-      // Listen for process exit
-      queueEvent(process.getPid(), Kevent.EVFILT_PROC, Kevent.EV_ADD | Kevent.EV_ONESHOT, Kevent.NOTE_EXIT | Kevent.NOTE_EXITSTATUS | Kevent.NOTE_REAP,
-                 pidPointer);
-      // Listen for self-generated signal to tell processor to wake-up and handling pending stdin requests
-      queueEvent(LibC.SIGUSR2, Kevent.EVFILT_SIGNAL, Kevent.EV_ADD, 0, pidPointer);
+      KeventArray keventArray = keventArrays.get();
```

Comment:
This is going to get called by the user's thread -- the one calling ``NuProcessBuilder.start()`` -- which is going to create a new ThreadLocal array of 64 kevents.  This is really inefficient compared to the kevent pool that was used.  Especially if the threads launching processes are one-shot.

We know we only need 4 kevent objects here, so why not construct what we need and skip the ThreadLocal altogether?